code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -105,6 +105,18 @@
 	..(gibbed)
 	regenerate_icons()
 
+/mob/living/simple_animal/pet/dog/corgi/RangedAttack(atom/A, params)
+	if(inventory_back)
+		inventory_back.afterattack(A, src, )
+
+/mob/living/simple_animal/pet/dog/corgi/UnarmedAttack(atom/A)
+	if(inventory_back)
+		if(istype(inventory_back, /obj/item/extinguisher))
```

Comment:
`istype()` works as a nullcheck as well, so you can remove this outer `if(inventory_back)` check as well.